integration/tests/overlay-focus-trap.test.js

```diff
@@ -0,0 +1,151 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync, nextRender, oneEvent, tabKeyDown } from '@vaadin/testing-helpers';
+import './not-animated-styles.js';
+import '@vaadin/button/vaadin-button.js';
+import '@vaadin/overlay/vaadin-overlay.js';
+import '@vaadin/radio-group/vaadin-radio-group.js';
+import '@vaadin/text-field/vaadin-text-field.js';
+import { getFocusableElements, isElementFocused } from '@vaadin/component-base/src/focus-utils.js';
+
+describe('focus-trap', () => {
+  let overlay, focusableElements;
+
+  function getFocusedElementIndex() {
+    return focusableElements.findIndex(isElementFocused);
+  }
+
+  describe('elements in light DOM', () => {
+    beforeEach(async () => {
+      overlay = fixtureSync('<vaadin-overlay focus-trap></vaadin-overlay>');
+      await nextRender();
```

Comment:
This line isn't needed per se, but I generally add it to reduce Lit conversion changes in the future.